aws-lc-rs/src/agreement.rs

```diff
@@ -368,10 +434,80 @@ impl EphemeralPrivateKey {
     }
 }
 
+impl AsDer<EcPrivateKeyRfc5915Der<'static>> for PrivateKey {
+    /// Serializes the key as a DER-encoded `ECPrivateKey` (RFC 5915) structure.
+    ///
+    /// X25519 is not supported.
+    ///
+    /// # Errors
+    /// `error::Unspecified`  if serialization failed.
+    fn as_der(&self) -> Result<EcPrivateKeyRfc5915Der<'static>, Unspecified> {
+        if AlgorithmID::X25519 == self.inner_key.algorithm().id {
+            return Err(Unspecified);
+        }
+
+        let mut outp = null_mut::<u8>();
+        let ec_key = {
+            ConstPointer::new(unsafe {
+                EVP_PKEY_get0_EC_KEY(self.inner_key.get_evp_pkey().as_const_ptr())
```

Comment:
Should this use EVP_PKEY_get1_EC_KEY/EC_KEY_free so the ref count is incremented? In theory AWS-LC could free this key since get0 doesn't increment the ref count. Disclaimer: I'm also not sure how Rust + aws-lc-rs play with threads and if anything can free `PrivateKey` concurrently to this call.

Reply:
Actually I think get0 makes sense because the overall EVP key will live as long as this function.